backend/app/services/knowledge/notification_service.py

```
@@ -0,0 +1,79 @@
+            detail_url=f"{settings.FRONTEND_URL}/knowledge/{kb.id}",
+        )
+
+        await webhook_notification_service.send_notification(notification)
+        logger.info(
+            f"Sent {event_type.value} notification to user {target_user.user_name} "
+            f"for knowledge base {kb.id}"
+        )
+    except Exception as e:
+        # Log error but don't fail the operation
+        logger.error(
+            f"Failed to send permission notification: {str(e)}",
```

Ruff (0.14.14)

Use explicit conversion flag

Replace with conversion flag

(RUF010)